File: uta-repo/snippets/swift/verify.swift
// UTA — Swift example: verify a credential via public HTTP API.
// Repo: https://github.com/alicelabs-llc/universal-trust-adapter
import Foundation

let card = CommandLine.arguments.count > 1 ? CommandLine.arguments[1] : ""
let url = URL(string: "https://www.marketnow.site/api/trust?action=verify")!
var req = URLRequest(url: url)
req.httpMethod = "POST"
req.setValue("application/json", forHTTPHeaderField: "Content-Type")
req.httpBody = try? JSONSerialization.data(withJSONObject: ["card": card])

let semaphore = DispatchSemaphore(value: 0)
URLSession.shared.dataTask(with: req) { data, _, _ in
    if let data, let json = try? JSONSerialization.jsonObject(with: data) as? [String: Any] {
        print("Decision:   \(json["decision"] ?? "?")")
        print("Format:     \(json["detected_format"] ?? "?")")
        print("Issuer:     \(json["issuer"] ?? "?")")
    }
    semaphore.signal()
}.resume()
semaphore.wait()

File: uta-repo/snippets/deno/verify.ts
/**
 * UTA — Deno example: verify a credential using native fetch (zero deps).
 * Repo: https://github.com/alicelabs-llc/universal-trust-adapter
 * Run:  deno run --allow-net verify.ts <card-string>
 */
const card = Deno.args[0];
if (!card) {
  console.error('Usage: deno run --allow-net verify.ts <card-string>');
  Deno.exit(1);
}
const resp = await fetch('https://www.marketnow.site/api/trust?action=verify', {
  method: 'POST',
  headers: { 'Content-Type': 'application/json' },
  body: JSON.stringify({ card }),
});
const result = await resp.json();
console.log('Decision:  ', result.decision);
console.log('Format:    ', result.detected_format);
console.log('Issuer:    ', result.issuer);
if (result.failed_stage) console.log('Failed at: ', result.failed_stage);
Deno.exit(result.decision === 'PERMIT' ? 0 : result.decision === 'DENY' ? 1 : 2);
